app/controllers/users.rb

```diff
@@ -0,0 +1,40 @@
+class MakersBnb < Sinatra::Base
+  delete '/' do 
+    session.clear
+    redirect ('/')
+  end 
+
+  post '/user/new' do
+    user = User.create(first_name: params[:first_name],
+                       last_name: params[:last_name],
+                       email: params[:email],
+                       password: params[:password],
+                       password_confirmation: params[:password_confirmation])
+    if user.saved? 
+      session[:user] = user.id
+      send_mail
```

Comment:
This is a prime example for a service (in this case, a Mailer object). It has a clear responsibility: why not wrap it in to a class?

```ruby
SendMail.with(user, :welcome)
```